app/application.js

```diff
@@ -27,80 +27,39 @@ const WINDOW_MIN_HEIGHT = 400;
 export const Platforms = {
   Mac: 1,
   Windows: 2,
-  Linux: 3,
-  isMac: (platform) => {
-    return platform === Platforms.Mac;
-  },
+  Linux: 3
 };
 
-export class DesktopApplication {
-
-  static instance = null;
-
-  static createSharedApplication() {
-    const getPlatform = () => {
-      const processPlatform = process.platform;
-      if (processPlatform === 'darwin') {
-        return Platforms.Mac;
-      } else if (processPlatform === 'win32') {
-        return Platforms.Windows;
-      } else if (processPlatform === 'linux') {
-        return Platforms.Linux;
-      }
-    };
-    this.instance = new DesktopApplication({
-      platform: getPlatform()
-    });
+function determinePlatform(platform) {
+  switch (platform) {
+    case "darwin":
+      return Platforms.Mac;
+    case "win32":
+      return Platforms.Windows;
+    case "linux":
+      return Platforms.Linux;
+    default:
+      throw `Unknown platform: ${platform}.`;
   }
+}
 
-  static sharedApplication() {
-    if (!this.instance) {
-      throw 'Attempting to access application before creation';
-    }
-    return this.instance;
-  }
+class DesktopApplication {
 
-  constructor({
-    platform
-  }) {
-    this.platform = platform;
-    this.isMac = Platforms.isMac(this.platform);
+  constructor() {
     app.setName(AppName);
-    this.createExtensionsServer();
+    this.platform = determinePlatform(process.platform);
+    this.isMac = this.platform === Platforms.Mac;
+    Store.set(StoreKeys.ExtServerHost, createExtensionsServer());
```

Comment:
Could we separate this into two lines:

```
const port = createExtensionsServer();
Store.set(StoreKeys.ExtServerHost, port);
```

Feels a bit too compressed to hide such an impactful function in the equivalent of `localStorage.setItem('port', createImportantServer());`